src/plugins/orangehrmOpenidAuthenticationPlugin/Controller/OpenIdConnectRedirectController.php

```diff
@@ -18,36 +18,160 @@
+                $this->getSocialMediaAuthenticationService()->getScope(),
+                $this->getSocialMediaAuthenticationService()->getRedirectURL()
+            );
+
+            $oidcClient->authenticate();
+
+            $userCredentials = new UserCredential();
+            $userCredentials->setUsername($oidcClient->requestUserInfo('email'));
+            $user = $this->getSocialMediaAuthenticationService()->getOIDCUser($userCredentials);
+
+
+            if (empty($user)) {
+                $this->getAuthUser()->addFlash(
+                    AuthUser::FLASH_LOGIN_ERROR,
+                    [
+                        'error' => AuthenticationException::UNEXPECT_ERROR,
+                        'message' => $this->getI18NHelper()->transBySource('No User Found'),
+                    ]
+                );
+                return new RedirectResponse($loginUrl);
+            }
+
+            if (sizeof($user) != 1) {
+                $this->getAuthUser()->addFlash(
+                    AuthUser::FLASH_LOGIN_ERROR,
+                    [
+                        'error' => AuthenticationException::UNEXPECT_ERROR,
+                        'message' => $this->getI18NHelper()->transBySource('Unexpected Error!'),
+                    ]
+                );
+                return new RedirectResponse($loginUrl);
+            }
+
+            $user = reset($user);
+            if (!($user instanceof User)) {
+                $this->getAuthUser()->addFlash(
+                    AuthUser::FLASH_LOGIN_ERROR,
+                    [
+                        'error' => AuthenticationException::UNEXPECT_ERROR,
+                        'message' => $this->getI18NHelper()->transBySource('Unexpected Error!'),
+                    ]
+                );
+                return new RedirectResponse($loginUrl);
+            }
+
+            if (!$user->getStatus()) {
+                $this->getAuthUser()->addFlash(
+                    AuthUser::FLASH_LOGIN_ERROR,
+                    [
+                        'error' => AuthenticationException::USER_DISABLED,
+                        'message' => $this->getI18NHelper()->transBySource('User Disabled'),
+                    ]
+                );
+            }
+
+            $success = $this->getSocialMediaAuthenticationService()->handleOIDCAuthentication($user);
+
+            if ($success) {
+                $this->getSocialMediaAuthenticationService()->setOIDCUserIdentity($user, $authProvider);
+                $this->getAuthUser()->setIsAuthenticated($success);
+                $this->getLoginService()->addOIDCLogin($user);
+            }
+
+            $redirectUrl = $this->handleSessionTimeoutRedirect();
+            if ($redirectUrl) {
+                return new RedirectResponse($redirectUrl);
+            }
+
+            $homePagePath = $this->getHomePageService()->getHomePagePath();
+            return $this->redirect($homePagePath);
+        } catch (AuthenticationException | Exception $e) {
```

Comment:
```suggestion
        } catch (Throwable $e) {
```
refer `ValidateController` and log errors and stack trace to the error log.